### DISCUSSION TITLE
Add port/fork for Cloudflare Pages

### DISCUSSION THREAD

**Discussion by henkisdabro** [gethyas/hyas#304] [Ideas]
Would be great to see this great project ported/forked over to support Cloudflare Pages (in addition to Netifly), with similar features such as _headers and _redirects and _functions and the new /functions/ folder and supported functions within. Support for preview environments, environment variables are all available, so it seems a feasible port that would be beneficial for us deeply invested in Cloudflare and wish to use this platform. I'm not that familiar with modifying the npm build process to ensure that Cloudflare's build process properly generate the correct files and ensure they fall into correct folders in the right format and so forth.

General documentation for Pages:
https://developers.cloudflare.com/pages/platform/

Existing migration guide from Netifly to Cloudflare Pages that might be useful:
https://developers.cloudflare.com/pages/migrations/migrating-from-netlify/

**Comment by h-enk** [comment]
That's absolutely interesting. I'll be giving this some thought on how to go about this.

**Comment by henkisdabro** [comment]
Awesome, glad you like the idea. I did a quick test deployment just now into Cloudflare Pages, setting the ENV’s for node, npm and hugo versions inside the CF Pages dashboard and it seems to build nicely already using the defaults. Even the `_redirects` and `_headers` files are building and ending up in the right folders already actually. Perhaps the `functions` folder needs adjustment due to Cloudflare using Workers for these. The netifly sitemap-submit plug-in will need to be rebuilt using CF workers if it’s still considered critical. Apart from this basic test I’d need your expertise :-)

**Reply by henkisdabro** [reply]
Just sharing a quick update that it's possible to circumvent needing to set `ENV` variables for the node.js+package versions on the cloudflare pages dashboard by simply adding a simple `.node-version` file in the root folder of the project (I used `16.15.0` similar to the requirements I found within`netifly.toml` file). This builds successfully.

I found information about this [here](https://developers.cloudflare.com/pages/platform/build-configuration/)
